### PR TITLE
[xvideos] Added support for quickies

### LINKED ISSUE #6356: Quickies on Xvideos
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm requesting a site-specific feature
- [X] I've verified that I'm running yt-dlp version **2023.02.17** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Worldwide

### Example URLs

https://www.xvideos.com/profiles/firolix#quickies/a/51392867
https://www.xvideos.com/profiles/firolix#quickies/a/63386619

### Provide a description that is worded well enough to be understood

Quickies, the version of shorts on Xvideos, aren't currently downloadable. Can you add support for them?

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', '-f', 'mp4', 'https://www.xvideos.com/profiles/firolix#quickies/a/51392867']
[debug] User config "C:\Users\Noah Bangs\yt-dlp.conf": ['-o', 'H:\\youtube-dl\\Export\\%(webpage_url_domain)s\\%(title)s - %(channel,artist - album)s.%(ext)s', '--retries', 'infinite', '--download-archive', 'registry.txt']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2023.02.17 [a0a7c01] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.19044-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg git-2020-04-03-52523b6, ffprobe git-2020-04-03-52523b6
[debug] Optional libraries: Cryptodome-3.17, brotli-1.0.9, certifi-2022.12.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Loaded 1782 extractors
[debug] Loading archive file 'registry.txt'
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2023.02.17, Current version: 2023.02.17
yt-dlp is up to date (2023.02.17)
[generic] Extracting URL: https://www.xvideos.com/profiles/firolix#quickies/a/51392867
[generic] 51392867: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] 51392867: Extracting information
[debug] Looking for embeds
ERROR: Unsupported URL: https://www.xvideos.com/profiles/firolix#quickies/a/51392867
Traceback (most recent call last):
  File "yt_dlp\YoutubeDL.py", line 1502, in wrapper
  File "yt_dlp\YoutubeDL.py", line 1578, in __extract_info
  File "yt_dlp\extractor\common.py", line 693, in extract
  File "yt_dlp\extractor\generic.py", line 2509, in _real_extract
yt_dlp.utils.UnsupportedError: Unsupported URL: https://www.xvideos.com/profiles/firolix#quickies/a/51392867
```

### PR DESCRIPTION
**Pull request by Yakabuff**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Adds support for the quickies videos type by converting it to a standard xvideos URL scheme

Fixes #6356

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

**Comment by pukkandan** [comment]
This is not the right approach. Add a new extractor matching only this URL pattern and returns `self.url_result(fixed_url)`

**Comment by pukkandan** [comment]
Implementation will roughly look like:
```
class XVideosQuickiesIE(InfoExtractor):
    IE_NAME = 'xvideos:quickies'
    _VALID_URL = r'https?://(?P<domain>(?:[^/]+\.)?xvideos2?\.com)/amateur-channels/[^#]+#quickies/a/(?P<id>\w+)'
    _TESTS = [...]
    def _real_extract(url):
        domain, id_ = self._match_valid_url(url).group('domain', 'id')
        return self.url_result(f'https://{domain}/video{id_}/_', XVideosIE, id)
```

**Comment by Yakabuff** [comment]
Fixed

File: yt_dlp/extractor/xvideos.py [hunk 1]
```diff
@@ -157,3 +157,24 @@ def _real_extract(self, url):
             'thumbnails': thumbnails,
             'age_limit': 18,
         }
+
+
+class XVideosQuickiesIE(InfoExtractor):
+    IE_NAME = 'xvideos:quickies'
+    _VALID_URL = r'https?://(?P<domain>(?:[^/]+\.)?xvideos2?\.com)/amateur-channels/[^#]+#quickies/a/(?P<id>\w+)'
```
Comment (pukkandan) [inline yt_dlp/extractor/xvideos.py:164]: I gave `\w+` in example code since idk how the ids can be. But looking at above extractor, it should be `\d+`?

**Review by pukkandan — APPROVED** [review]

File: yt_dlp/extractor/xvideos.py [hunk 2]
```diff
@@ -157,3 +157,24 @@ def _real_extract(self, url):
...
+            'title': 'Verification video',
+            'age_limit': 18,
+            'duration': 16,
+            'thumbnail': r're:^https://cdn.*-pic.xvideos-cdn.com/.+\.jpg',
+        }
+    }]
+
+    def _real_extract(self, url):
+        domain, id_ = self._match_valid_url(url).group('domain', 'id')
+        return self.url_result(f'https://{domain}/video{id_}/_', XVideosIE, id)
```
Comment (bashonly) [inline yt_dlp/extractor/xvideos.py:180]:
```suggestion
        return self.url_result(f'https://{domain}/video{id_}/_', XVideosIE, id_)
```

File: yt_dlp/extractor/xvideos.py (hunk 1 above)
Comment (pukkandan) [reply yt_dlp/extractor/xvideos.py:164]:
```suggestion
    _VALID_URL = r'https?://(?P<domain>(?:[^/]+\.)?xvideos2?\.com)/amateur-channels/[^#]+#quickies/a/(?P<id>\d+)'
```

**Comment by Menard01** [comment]
It seems to be broken again

**Comment by pukkandan** [comment]
Comments are not a substitute for proper issue reporting

**Pull request merged**